docs/dqx/docs/reference/quality_rules.mdx

```diff
@@ -470,6 +470,116 @@ checks = [
 ```
 </details>
 
+### Usage examples of check functions with Map and Array columns
+
+You can apply checks to complex columns (e.g. `MapType` or `ArrayType`) by passing column expressions to the check function.
+
+Example checks defined in YAML:
+
+<details style={{ backgroundColor: 'transparent', color: 'neutral' }}>
+<summary>**Checks on complex columns defined in YAML**</summary>
+```yaml
+# map_value_is_not_null check
+- criticality: error
+  check:
+    function: is_not_null
+    arguments:
+      col_name: col("map_col").getItem("map_key")
+
+# array_element_is_not_null check
+- criticality: error
+  check:
+    function: is_not_null
+    arguments:
+      col_name: try_element_at("array_col", 1)
+
+# all_array_elements_less_than check
+- criticality: error
+  check:
+    function: is_not_greater_than
+    arguments:
+      col_name: array_max("array_col")
+      limit: current_date()
+
+# array_contains_element check
+- criticality: error
+  check:
+    function: is_not_less_than
+    arguments:
+      col_name: array_position("array_col", "search_value")
+      limit: 1
+```
+</details>
+
+<details style={{ backgroundColor: 'transparent', color: 'neutral' }}>
+<summary>**Checks defined on complex columns defined using DQX classes**</summary>
+```python
+import pyspark.sql.functions as F
+from databricks.labs.dqx.rule import DQRule
+from databricks.labs.dqx.col_functions import *
+
+checks = [
+    DQRule(
+        criticality="error",
+        check=is_not_null(F.col("map_col").getItem("map_key"))
+    ),
+    DQRule(
+        criticality="error",
+        check=is_not_null(F.try_element_at("array_col", 1))
+    ),
+    DQRule(
+        criticality="error",
+        check=is_not_greater_than(F.array_max("array_col"), F.current_date())
+    ),
+    DQRule(
+        criticality="error",
+        check=is_not_less_than(F.array_position("array_col", "search_value"), 1)
+    )
+]
+```
+</details>
+
+<details style={{ backgroundColor: 'transparent', color: 'neutral' }}>
+<summary>**Checks defined on complex columns defined in YAML using SQL expressions**</summary>
+```python
+checks = [
+    {
+        "criticality": "error",
+        "name": "col_map_element_at_col1_key1_is_not_greater_than_10",
+        "check": {"function": "sql_expression", "arguments": {"expression": "element_at(col1, 'key1') < 10"}},
+    },
+    {
+        "criticality": "error",
+        "name": "col_array_element_at_col1_key1_is_not_greater_than_10",
```

Comment:
I would provide a comment similar to other checks. The name has the description but it's harder to read.